src/pages/iou/request/step/withWritableReportOrNotFound.tsx

```diff
@@ -0,0 +1,50 @@
+import type {RouteProp} from '@react-navigation/core';
+import type {ComponentType, ForwardedRef} from 'react';
+import React, {forwardRef} from 'react';
+import type {OnyxEntry} from 'react-native-onyx';
+import {withOnyx} from 'react-native-onyx';
+import FullPageNotFoundView from '@components/BlockingViews/FullPageNotFoundView';
+import getComponentDisplayName from '@libs/getComponentDisplayName';
+import type {MoneyRequestNavigatorParamList} from '@libs/Navigation/types';
+import * as ReportUtils from '@libs/ReportUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import type {Report} from '@src/types/onyx';
+
+type WithWritableReportOrNotFoundOnyxProps = {
+    /** The report corresponding to the reportID in the route params */
+    report: OnyxEntry<Report>;
+};
+
+type WithWritableReportOrNotFoundProps = WithWritableReportOrNotFoundOnyxProps & {route: RouteProp<MoneyRequestNavigatorParamList, typeof SCREENS.MONEY_REQUEST.STEP_WAYPOINT>};
```

Comment:
Since this route will get more complex when other screens which uses this HOC will be migrated, I'd put it into separate type, so
```suggestion
type Route = RouteProp<MoneyRequestNavigatorParamList, typeof SCREENS.MONEY_REQUEST.STEP_WAYPOINT>;

type WithWritableReportOrNotFoundProps = WithWritableReportOrNotFoundOnyxProps & {route: Route};
```
What do you think?